plugins/modules/k8s_drain.py

```diff
@@ -0,0 +1,394 @@
+        return "timeout reached while pods were still running."
+
+    def evict_pods(self, pods):
+        for namespace, name in pods:
+            definition = {
+                'metadata': {
+                    'name': name,
+                    'namespace': namespace
+                }
+            }
+            if self._delete_options:
+                definition.update({'delete_options': self._delete_options})
+            try:
+                if self._drain_options.get('disable_eviction'):
+                    body = V1DeleteOptions(**definition)
+                    self._api_instance.delete_namespaced_pod(name=name, namespace=namespace, body=body)
+                else:
+                    body = V1beta1Eviction(**definition)
+                    self._api_instance.create_namespaced_pod_eviction(name=name, namespace=namespace, body=body)
+                self._changed = True
+            except ApiException as exc:
+                if exc.reason != "Not Found":
+                    self._module.fail_json(msg="Failed to delete pod {0}/{1} due to: {2}".format(namespace, name, exc.reason))
+            except Exception as exc:
+                self._module.fail_json(msg="Failed to delete pod {0}/{1} due to: {2}".format(namespace, name, to_native(exc)))
+
+    def delete_or_evict_pods(self, node_unschedulable):
+        try:
+            field_selector = "spec.nodeName={name}".format(name=self._module.params.get('name'))
+            pod_list = self._api_instance.list_pod_for_all_namespaces(field_selector=field_selector)
+            # Filter pods
+            force = self._drain_options.get('force', False)
+            ignore_daemonset = self._drain_options.get('ignore_daemonsets', False)
+            pods, warnings, errors = filter_pods(pod_list.items, force, ignore_daemonset)
+            if errors:
+                self._module.fail_json(msg="Pod deletion errors: {0}".format(" ".join(errors)))
+        except ApiException as exc:
+            if exc.reason != "Not Found":
+                self._module.fail_json(msg="Failed to list pod from node {name} due to: {reason}".format(
+                                       name=self._module.params.get('name'), reason=exc.reason), status=exc.status)
+            pods = []
+        except Exception as exc:
+            self._module.fail_json(msg="Failed to list pod from node {name} due to: {error}".format(
+                                   name=self._module.params.get('name'), error=to_native(exc)))
+
+        # Mark node as unschedulable
+        result = []
+        if not node_unschedulable:
+            self.patch_node(unschedulable=True)
+            result.append("node {0} marked unschedulable.".format(self._module.params.get('name')))
+            self._changed = True
+        else:
+            warnings.append("node {0} already marked unschedulable.".format(self._module.params.get('name')))
+
+        # Delete Pods
+        if pods:
+            self._changed = True
```

Comment:
This won't necessarily be true. Since this is set correctly in the `evict_pods` method, I would delete it here.